starlite/signature.py

```diff
@@ -60,63 +76,247 @@ def parse_values_from_connection_kwargs(
+                f"or provided dependency."
+            )
+
+    def collect_dependency_names(self, parameter: SignatureParameter) -> None:
+        """
+        Add parameter name of dependencies declared using `Dependency()` function to the set of all
+        dependency names.
+
+        Parameters
+        ----------
+        parameter : SignatureParameter
+        """
+        if is_dependency_field(parameter.default):
+            self.dependency_name_set.add(parameter.name)
+
+    def record_default(self, parameter: SignatureParameter) -> None:
+        """
+        If `parameter` has defined default, map it to the parameter name in `self.defaults`.
+
+        Parameters
+        ----------
+        parameter : SignatureParameter
+        """
+        if parameter.default_defined:
+            self.defaults[parameter.name] = parameter.default
+
+    def get_type_annotation_from_plugin(self, parameter: SignatureParameter, plugin: PluginProtocol) -> Any:
+        """
+        Use plugin declared for parameter annotation type to generate a pydantic model.
+
+        Parameters
+        ----------
+        parameter : SignatureParameter
+        plugin : PluginProtocol
+
+        Returns
+        -------
+        Any
+        """
+        type_args = get_args(parameter.annotation)
+        type_value = type_args[0] if type_args else parameter.annotation
+        self.field_plugin_mappings[parameter.name] = PluginMapping(plugin=plugin, model_class=type_value)
+        pydantic_model = plugin.to_pydantic_model_class(model_class=type_value)
+        if type_args:
+            return List[pydantic_model]  # type:ignore[valid-type]
+        return pydantic_model
+
+    @staticmethod
+    def field_definition_from_parameter(parameter: SignatureParameter) -> Tuple[Any, Any]:
+        """
+        Construct an `(<annotation>, <default>)` tuple, appropriate for `pydantic.create_model()`.
+
+        Parameters
+        ----------
+        parameter : SignatureParameter
+
+        Returns
+        -------
+        tuple[Any, Any]
+        """
+        if parameter.default_defined:
+            field_definition = (parameter.annotation, parameter.default)
+        elif not parameter.optional:
+            field_definition = (parameter.annotation, ...)
+        else:
+            field_definition = (parameter.annotation, None)
+        return field_definition
+
+    @property
+    def signature_parameters(self) -> Generator[SignatureParameter, None, None]:
+        """
+        Iterable of `SignatureModel` instances, that represent the parameters of the function
+        signature that should be included in the `SignatureModel` type.
+
+        Returns
+        -------
+        Generator[SignatureParameter, None, None]
+        """
+        for name, parameter in self.signature.parameters.items():
+            if name in self.SKIP_NAMES:
+                continue
+            yield SignatureParameter.new(self.fn_name, name, parameter)
```

Comment:
Why use new rather than a standard constructor?

Reply:
If I think of a good reason I'll let you know